packages/api/amplify_api/test/graphql_test.dart

```diff
@@ -452,6 +449,47 @@ void main() {
 
       await mockWebSocketBloc!.done.future;
     });
+
+    test(
+        'should reconnect after 15 seconds when appsync ping fails multiple times',
+        () async {
+      final blocReady = Completer<void>();
+
+      expect(
+        hubEvents,
+        emitsInOrder(
+          [
+            disconnectedHubEvent,
+            connectingHubEvent,
+            connectedHubEvent,
+            connectingHubEvent,
+            connectingHubEvent,
+            connectedHubEvent,
+          ],
+        ),
+      );
+
+      initMockConnection(
+        mockWebSocketBloc!,
+        mockWebSocketService!,
+        subscriptionRequest.id,
+      );
+
+      Amplify.API.subscribe(
+        subscriptionRequest,
+        onEstablished: blocReady.complete,
+      );
+
+      await blocReady.future;
+
+      fakePlatform.controller.sink.add(ConnectivityResult.wifi);
+
+      _failClient = true;
+
+      await Future<void>.delayed(const Duration(seconds: 15));
```

Comment:
Not sure I follow the 15 seconds - you have a poll interval of 1s and a `_maxFailAttempts = 5` so theoretically only 5s is needed here.

Reply:
Can you add a test where `GET /ping` succeeds but returns `400` or a non-`healthy` body.

Reply:
15 seconds is tied to the retry strategy duration, not the poll interval. 15 seconds is enough time for 5 exponential back off attempts. I've adjusted the title of the test to reflect this.